File: src/Samples/Net_Framework/JSON_Serializer/InteropHelper.vb

Namespace Global.Newtonsoft.Json
    Public Module InteropHelper
#If Not BRIDGE Then
        <JSIL.Meta.JSReplacement("$value")>
        Public Function Unbox(ByVal value As Object) As Object
#Else
        <Bridge.Template("({value}.v != undefined ? {value}.v : {value})")>
        Public Function Unbox(ByVal value As Object) As Object
#End If

            Return value
        End Function
    End Module
End Namespace

File: src/Samples/Net_Framework/JSON_Serializer/InteropHelper.fs
namespace Newtonsoft.Json

#nowarn "0044" // Suppress warning FS0044 (about deprecated JSIL.Meta.JSReplacement)

module InteropHelper =

#if !BRIDGE
    [<JSIL.Meta.JSReplacement("$value")>]
#else
    [<Bridge.Template("({value}.v != undefined ? {value}.v : {value})")>]
#endif
    let Unbox (value : obj) =
        value
